mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MessageMapper.java

```diff
@@ -197,7 +198,31 @@ long countUnseenMessagesInMailbox(Mailbox mailbox)
          * Fetch the complete {@link MailboxMessage}
          * 
          */
-        Full
+        Full;
+
+        static FetchType reduce(FetchType fetchType1, FetchType fetchType2) {
```

Comment:
do you really find that simpler than the previous code?

Reply:
Yes I do.

I believe  a scala syntax would make this easier.